src/com/mqtt_paho/MQTTClient.cpp

```diff
@@ -186,6 +187,18 @@ int CMQTTClient::initClient() {
   mClientConnectionOptions.onSuccess = onMqttConnectionSucceed;
   mClientConnectionOptions.onFailure = onMqttConnectionFailed;
   mClientConnectionOptions.context = this;
+
+  if ("" != gMqttClientConfigFile) { //file was provided
+    CMQTTClientConfigFileParser::MQTTConfigFromFile result = CMQTTClientConfigFileParser::MQTTConfigFromFile(mUsername, mPassword);
```

Comment:
do you really need user name and password as fields of the Client class. You are immediately forwarding them to the client connection options. So wouldn't it be nicer to have them as local variables?